### PR TITLE
Resolves #22 - Setup routing with React Router 4

### LINKED ISSUE #22: Setup routing React Router 4
### Description 
Set-up RR v4

### Ramp up tools
- https://reacttraining.com/react-router/

### PR DESCRIPTION
**Pull request by andela-ksolomon**
Resolves #22 

Description:
Setup routing using react-router 4, currently we have the index route linked to the sign in page and the `/my-activities` linked to the activities page

Fix:
 - Add the routes.jsx file that contains all routes and their respective components
 - Implemented the historyfallbackApi to ensure all routes fallbacks to the index.html - Webpack DevServer

How to test.
Pull this branch if you don't have it already. (git pull origin ch-setup-react-router-155749607)
Run yarn start. The signIn page should appear.
Try navigate to `/my-activities` you should see the Activities Page

### CODE REVIEW

**Comment by john555** [comment]
LGTM

File: src/index.jsx [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
-import App from './components/App';
+import Routes from './routes';
 import './assets/scss/style.scss';
 
 const { NODE_ENV } = process.env;
 
 const mount = (Component) => {
   render(
-    <div>
+    <BrowserRouter>
```
Comment (john555) [inline src/index.jsx]: You could move `BrowseRouter` to `routes.js` as the top most element.

File: package.json [hunk 2]
```diff
@@ -83,4 +85,4 @@
       "!**/node_modules/**"
     ]
   }
-}
+}
```
Comment (john555) [inline package.json]: Add a new line at the end of each file.

File: src/routes.jsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import App from './components/App';
+import Signin from './containers/SignIn';
+
+/**
+ * @name Routes
+ * @summary Renders the application routes
+ * @return {jsx} React node for the application routes
+ */
+const Routes = () => (
+  <div>
```
Comment (john555) [inline src/routes.jsx]: This could be browser router.

Comment (andela-ksolomon) [reply src/routes.jsx]: A variable already exist with the name browser router, I think router would be fine as well. What do you think?

**Comment by john555** [comment]
@andela-ksolomon Is router a component? If so, you could place it in the components folder.

**Comment by andela-ksolomon** [comment]
@john555 I did not place it in the components folder initially because router is a top level components wrapping other parent components

**Comment by andela-ksolomon** [comment]
@john555 changes has been made

**Comment by john555** [comment]
Since the router is a top-level component, could it be moved to the containers folder?

**Pull request merged**